### PR TITLE
Added README to backward compat.

### PR DESCRIPTION
**Pull request by fulmicoton**
Backward compatibility README

### CODE REVIEW

File: quickwit-backward-compat/README.md [hunk 1]
```diff
@@ -3,7 +3,50 @@
 This is just a project used to test backward compatibility of Quickwit.
 Right now, only SplitMetadata is tested.
 
-The build.rs script autocreates json sample files when something changes in the format.
-A unit test then checks that all of these sample files deserialize to the right
-`SplitMetadataAndFooterOffsets` object.
+We want to be able to read all past version of SplitMetadata, but only write the most recent format.
+The tests consists in pairs of json file.
+XXXX.json and XXXX.expected.json
```
Comment (PSeitz) [inline quickwit-backward-compat/README.md:8]: wasn't it like?

vX-[sha].json

File: quickwit-backward-compat/README.md [hunk 2]
```diff
@@ -3,7 +3,50 @@
 This is just a project used to test backward compatibility of Quickwit.
 Right now, only SplitMetadata is tested.
 
-The build.rs script autocreates json sample files when something changes in the format.
-A unit test then checks that all of these sample files deserialize to the right
-`SplitMetadataAndFooterOffsets` object.
+We want to be able to read all past version of SplitMetadata, but only write the most recent format.
```
Comment (PSeitz) [inline quickwit-backward-compat/README.md:6]:
```suggestion
We want to be able to read all past versions of SplitMetadata, but only write the most recent format.
```

File: quickwit-backward-compat/README.md [hunk 3]
```diff
@@ -3,7 +3,50 @@
...
+The tests consists in pairs of json file.
+XXXX.json and XXXX.expected.json
+
+XXXX.json consists in a json file in a possibly old format.
```
Comment (PSeitz) [inline quickwit-backward-compat/README.md:10]: more details about "possibly" would be good

File: quickwit-backward-compat/README.md [hunk 4]
```diff
@@ -3,7 +3,50 @@
...
+struct.
+It is also possible to rename a field in a backward compatible manner
+by using the `#[serde(alias)]`.
+
+For this type of change it is not required to update the serialization version.
+
+When introducing such a change:
+- modify your model with the help of the attributes above.
+- commit the 2 files that were autogenerated by build.rs
+- fix the `.expected.json` that failed.
```
Comment (PSeitz) [inline quickwit-backward-compat/README.md:41]: This reads like just change the `.expected.json` :)

File: quickwit-backward-compat/README.md (hunk 1 above)
Comment (fulmicoton) [reply quickwit-backward-compat/README.md:8]: The sha1 is part of the XXXX. You are not required to include any sha1 in the name by the way. It is possible to add manually crafter use cases.

File: quickwit-backward-compat/README.md (hunk 3 above)
Comment (fulmicoton) [reply quickwit-backward-compat/README.md:10]:
```suggestion
XXXX.json consists in a json file in an old format.
```

**Comment by fulmicoton** [comment]
Abusing my power and merging this.

**Pull request merged**